cdc/sink/ddlsink/mysql/mysql_ddl_sink.go

```diff
@@ -306,23 +331,123 @@ func needFormatDDL(db *sql.DB, cfg *pmysql.Config) bool {
 	return false
 }
 
-// WriteCheckpointTs does nothing.
-func (m *DDLSink) WriteCheckpointTs(_ context.Context, _ uint64, _ []*model.TableInfo) error {
-	// Only for RowSink for now.
-	return nil
+func getDDLCreateTime(ctx context.Context, db *sql.DB) int64 {
+	var ddlCreateTime int64 = -1 // default when scan failed
+	row, err := db.QueryContext(ctx, "SELECT UNIX_TIMESTAMP()")
+	if err != nil {
+		log.Warn("selecting unix timestamp failed", zap.Error(err))
+	} else {
+		for row.Next() {
+			err = row.Scan(&ddlCreateTime)
+			if err != nil {
+				log.Warn("getting ddlCreateTime failed", zap.Error(err))
+			}
+		}
+		//nolint:sqlclosecheck
+		_ = row.Close()
+		_ = row.Err()
+	}
+	return ddlCreateTime
 }
 
-// Close closes the database connection.
-func (m *DDLSink) Close() {
-	if m.statistics != nil {
-		m.statistics.Close()
-	}
-	if m.db != nil {
-		if err := m.db.Close(); err != nil {
-			log.Warn("MySQL ddl sink close db wit error",
-				zap.String("namespace", m.id.Namespace),
-				zap.String("changefeed", m.id.ID),
-				zap.Error(err))
+var checkRunningSQL = `
```

Comment:
we can update the sql to directly selete the jobs whose create_time > ddlCreateTime, instead of loop the rows to check.

Besides, the sql can also add the clause like `AND JOB_TYPE LIKE "add index%%`